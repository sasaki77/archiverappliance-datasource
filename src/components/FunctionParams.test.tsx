import React from 'react';
import { render } from '@testing-library/react';

import { FunctionParams, FunctionParamsProps } from './FunctionParams';

const setup = (propOverrides?: object) => {
  const props: FunctionParamsProps = {
    func: {
      def: {
        category: 'Filter Series',
        defaultParams: [5, 'avg'],
        name: 'top',
        params: [
          {
            name: 'number',
            type: 'int',
          },
          {
            name: 'value',
            options: ['avg', 'min', 'max', 'absoluteMin', 'absoluteMax', 'sum'],
            type: 'string',
          },
        ],
      },
      params: ['5', 'avg'],
    },
    index: 1,
    onChange: jest.fn(),
    onRunQuery: jest.fn(),
  };

  Object.assign(props, propOverrides);

  return render(<FunctionParams {...props} />);
};

describe('Render', () => {
  it('should render component', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
