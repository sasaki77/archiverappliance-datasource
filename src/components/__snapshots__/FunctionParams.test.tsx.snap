// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      
      <div>
        <div
          class="aa-func-param"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-owns="react-autowhatever-1"
            class="react-autosuggest__container"
            role="combobox"
          >
            <input
              aria-autocomplete="list"
              aria-controls="react-autowhatever-1"
              autocomplete="off"
              class="tight-form-func-param"
              placeholder="number"
              spellcheck="false"
              style="width: 1ch; font-family: Consolas, "Courier New", Courier, Monaco, monospace;"
              type="text"
              value="5"
            />
            <div
              class="react-autosuggest__suggestions-container"
              id="react-autowhatever-1"
              role="listbox"
            />
          </div>
        </div>
      </div>
      <span
        class="comma"
      >
        , 
      </span>
      <div>
        <div
          class="aa-func-param"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-owns="react-autowhatever-1"
            class="react-autosuggest__container"
            role="combobox"
          >
            <input
              aria-autocomplete="list"
              aria-controls="react-autowhatever-1"
              autocomplete="off"
              class="tight-form-func-param"
              placeholder="value"
              spellcheck="false"
              style="width: 3ch; font-family: Consolas, "Courier New", Courier, Monaco, monospace;"
              type="text"
              value="avg"
            />
            <div
              class="react-autosuggest__suggestions-container"
              id="react-autowhatever-1"
              role="listbox"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    
    <div>
      <div
        class="aa-func-param"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="react-autowhatever-1"
          class="react-autosuggest__container"
          role="combobox"
        >
          <input
            aria-autocomplete="list"
            aria-controls="react-autowhatever-1"
            autocomplete="off"
            class="tight-form-func-param"
            placeholder="number"
            spellcheck="false"
            style="width: 1ch; font-family: Consolas, "Courier New", Courier, Monaco, monospace;"
            type="text"
            value="5"
          />
          <div
            class="react-autosuggest__suggestions-container"
            id="react-autowhatever-1"
            role="listbox"
          />
        </div>
      </div>
    </div>
    <span
      class="comma"
    >
      , 
    </span>
    <div>
      <div
        class="aa-func-param"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="react-autowhatever-1"
          class="react-autosuggest__container"
          role="combobox"
        >
          <input
            aria-autocomplete="list"
            aria-controls="react-autowhatever-1"
            autocomplete="off"
            class="tight-form-func-param"
            placeholder="value"
            spellcheck="false"
            style="width: 3ch; font-family: Consolas, "Courier New", Courier, Monaco, monospace;"
            type="text"
            value="avg"
          />
          <div
            class="react-autosuggest__suggestions-container"
            id="react-autowhatever-1"
            role="listbox"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
